tools/trace_analyzer_tool.cc

```diff
@@ -1504,322 +1582,176 @@ Status TraceAnalyzer::Handle(const WriteQueryTraceRecord& record,
   // process the reord that is committed. If write is non-transaction,
   // HasBeginPrepare()==false, so we process it normally.
   WriteBatch batch(record.GetWriteBatchRep().ToString());
-  if (batch.HasBeginPrepare() && !batch.HasCommit()) {
+  if (batch.Count() == 0 || (batch.HasBeginPrepare() && !batch.HasCommit())) {
+    if (result != nullptr) {
+      result->reset();
+    }
     return Status::OK();
   }
-  c_time_ = record.GetTimestamp();
+  write_result_.reset(
+      new TraceAnalysisResult(record.GetTimestamp(), record.GetTraceType()));
+  write_result_->Reserve(static_cast<size_t>(batch.Count()));
+
+  // write_result_ will be updated in batch's handler during iteration.
   Status s = batch.Iterate(this);
   if (!s.ok()) {
     fprintf(stderr, "Cannot process the write batch in the trace\n");
+    write_result_.reset();
+    if (result != nullptr) {
+      result->reset();
+    }
     return s;
   }
+  if (result != nullptr) {
+    // "Move" the temporary `write_result_` to `result`.
+    result->reset(write_result_.release());
+  }
   return Status::OK();
 }
 
 Status TraceAnalyzer::Handle(const GetQueryTraceRecord& record,
-                             std::unique_ptr<TraceRecordResult>* /*result*/) {
+                             std::unique_ptr<TraceRecordResult>* result) {
   total_gets_++;
-
-  uint32_t cf_id = record.GetColumnFamilyID();
-  Slice key = record.GetKey();
-  uint64_t ts = record.GetTimestamp();
-
-  Status s;
-  size_t value_size = 0;
-  if (FLAGS_convert_to_human_readable_trace && trace_sequence_f_) {
-    s = WriteTraceSequence(TraceOperationType::kGet, cf_id, key, value_size,
-                           ts);
-    if (!s.ok()) {
-      return Status::Corruption("Failed to write the trace sequence to file");
-    }
-  }
-
-  if (ta_[TraceOperationType::kGet].sample_count >= sample_max_) {
-    ta_[TraceOperationType::kGet].sample_count = 0;
-  }
-  if (ta_[TraceOperationType::kGet].sample_count > 0) {
-    ta_[TraceOperationType::kGet].sample_count++;
-    return Status::OK();
-  }
-  ta_[TraceOperationType::kGet].sample_count++;
-
-  if (!ta_[TraceOperationType::kGet].enabled) {
-    return Status::OK();
-  }
-  value_size = 10;
```

Comment:
@zhichao-cao  Why change value size to 10?

Reply:
In the Get trace record, we do not collect the value. So, we do not have the "value" part. In order to simplify the later on calculation, we just give a fixed value. Maybe it would be better to add a comments here to make it more clear

Reply:
ACK

Reply:
One more question, why we don't do the same thing for other key only operations (seeks, deletes, etc.)?

Reply:
obviously, we forgot to do the same thing for other non-value recorded operations........